test/one/profiler/test/TestDeclaration.java

```diff
@@ -0,0 +1,148 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class TestDeclaration {
+    private static final Logger log = Logger.getLogger(TestDeclaration.class.getName());
+
+    private final List<String> allDirs;
+    private final List<Pattern> includeGlobs;
+    private final List<Pattern> skipGlobs;
+
+    public TestDeclaration(List<String> allDirs, List<String> globs, List<String> skipGlobs) {
+        this.allDirs = allDirs;
+        this.includeGlobs = filterFromGlobs(globs);
+        this.skipGlobs = filterFromGlobs(skipGlobs);
+
+        log.log(Level.FINE, "Test Directories: " + allDirs);
+        log.log(Level.FINE, "Test Filters: " + globs);
+        log.log(Level.FINE, "Skip Filters: " + skipGlobs);
+    }
+
+    public static TestDeclaration parse(String[] args) {
+        // All available directories.
+        List<String> testDirectories = new ArrayList<>();
+
+        // Glob filters matching "ClassName.methodName".
+        List<String> filters = Arrays.asList(args);
+
+        List<String> skipFilters = new ArrayList<>();
+        String skipProperty = System.getProperty("skip");
+        if (skipProperty != null && !skipProperty.isEmpty()) {
+            for (String skip : skipProperty.split(" ")) {
+                skipFilters.add(skip);
+            }
+        }
+
+        List<String> allTestDirs = new ArrayList<>();
+        File[] files = new File("test/test").listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    allTestDirs.add(file.getName());
+                }
+            }
+        }
+
+        return new TestDeclaration(allTestDirs, filters, skipFilters);
+    }
+
+    private static List<RunnableTest> getRunnableTests(Class<?> cls) {
+        List<RunnableTest> rts = new ArrayList<>();
+        for (Method m : cls.getMethods()) {
+            for (Test t : m.getAnnotationsByType(Test.class)) {
+                rts.add(new RunnableTest(m, t));
+            }
+        }
+        return rts;
+    }
+
+    private static List<RunnableTest> getRunnableTests(String dir) {
+        String className = "test." + dir + "." + Character.toUpperCase(dir.charAt(0)) + dir.substring(1) + "Tests";
+        try {
+            return getRunnableTests(Class.forName(className));
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private List<RunnableTest> getRunnableTests(List<String> globs) {
+        return globs.stream().filter(g -> g.equals(g.toLowerCase())).flatMap(g -> getRunnableTests(g).stream()).sorted(Comparator.comparing(RunnableTest::testName)).collect(Collectors.toList());
```

Comment:
Not readable. Streams are usually formatted as follows:
```
return globs.stream()
        .filter(g -> g.equals(g.toLowerCase()))
        .flatMap(g -> getRunnableTests(g).stream())
        .sorted(Comparator.comparing(RunnableTest::testName))
        .collect(Collectors.toList());
```

Given that `List<String> globs` is always a list of directories here, `filter` can be removed.

Also, `getRunnableTests` method has too many overloads, I'd inline this one in `getRunnableTests()` below.